test/test_utils.py

```diff
@@ -268,6 +268,20 @@ def run_fn(tensor1, tensor2):
         out = checkpoint(run_fn, input_var, None)
         out.sum().backward()
 
+    def test_checkpoint_partial_grad(self):
+        def run_fn(tensor1, tensor2):
+            # tensor 2 is used for other application logic
+            return tensor1
+        input_var = torch.randn(1, 100, requires_grad=True)
+        input_var2 = torch.randn(1, 100, requires_grad=False)
+        out = checkpoint(run_fn, input_var, input_var2)
```

Comment:
This would work with the old version of the code no? The old code only fails if an output of the function does not require gradients right?